applications/TrilinosApplication/tests/cpp_tests/test_builder_and_solver.cpp

```diff
@@ -0,0 +1,893 @@
+            rModelPart.AddElement(Kratos::make_intrusive<TestBarElement>( 1, pgeom1, p_prop));
+            GeometryType::Pointer pgeom2 = Kratos::make_shared<Line2D2<NodeType>>(PointerVector<NodeType>{std::vector<NodeType::Pointer>({pnode2, pnode3})});
+            rModelPart.AddElement(Kratos::make_intrusive<TestBarElement>( 2, pgeom2, p_prop));
+        } else {
+            if (rank == 0) {
+                // Create nodes
+                auto pnode1 = rModelPart.CreateNewNode(1, 0.0, 0.0, 0.0);
+                pnode1->FastGetSolutionStepValue(PARTITION_INDEX) = 0;
+                auto pnode2 = rModelPart.CreateNewNode(2, 1.0, 0.0, 0.0);
+                pnode2->FastGetSolutionStepValue(PARTITION_INDEX) = 1;
+
+                // Create elements
+                GeometryType::Pointer pgeom1 = Kratos::make_shared<Line2D2<NodeType>>(PointerVector<NodeType>{std::vector<NodeType::Pointer>({pnode1, pnode2})});
+                rModelPart.AddElement(Kratos::make_intrusive<TestBarElement>( 1, pgeom1, p_prop));
+            } else if (rank == 1) {
+                // Create nodes
+                auto pnode2 = rModelPart.CreateNewNode(2, 1.0, 0.0, 0.0);
+                auto pnode3 = rModelPart.CreateNewNode(3, 2.0, 0.0, 0.0);
+
+                /// Add PARTITION_INDEX
+                for (auto& r_node : rModelPart.Nodes()) {
+                    r_node.FastGetSolutionStepValue(PARTITION_INDEX) = 1;
+                }
+
+                // Create elements
+                GeometryType::Pointer pgeom2 = Kratos::make_shared<Line2D2<NodeType>>(PointerVector<NodeType>{std::vector<NodeType::Pointer>({pnode2, pnode3})});
+                rModelPart.AddElement(Kratos::make_intrusive<TestBarElement>( 2, pgeom2, p_prop));
+            }
+        }
+
+        /// Add dof
+        for (auto& r_node : rModelPart.Nodes()) {
+            r_node.AddDof(DISPLACEMENT_X, REACTION_X);
+            r_node.AddDof(DISPLACEMENT_Y, REACTION_Y);
+            r_node.AddDof(DISPLACEMENT_Z, REACTION_Z);
+        }
+
+        /// Initialize elements
+        const auto& r_process_info = rModelPart.GetProcessInfo();
+        for (auto& r_elem : rModelPart.Elements()) {
+            r_elem.Initialize(r_process_info);
+            r_elem.InitializeSolutionStep(r_process_info);
+        }
+
+        // Set initial solution
+        for (auto& r_node : rModelPart.Nodes()) {
+            (r_node.FastGetSolutionStepValue(DISPLACEMENT)).clear();
+            (r_node.FastGetSolutionStepValue(DISPLACEMENT, 1)).clear();
+            (r_node.FastGetSolutionStepValue(DISPLACEMENT, 2)).clear();
+        }
+
+        // Fix dofs
+        for (auto& r_node : rModelPart.Nodes()) {
+            r_node.Fix(DISPLACEMENT_Y);
+            r_node.Fix(DISPLACEMENT_Z);
+        }
+        // Fix X in first node
+        if (rModelPart.HasNode(1)) {
+            auto pnode = rModelPart.pGetNode(1);
+            pnode->Fix(DISPLACEMENT_X);
+        }
+
+        // TODO: Fix this once MPC work!
```

Comment:
Instead of commenting I would skip the test

Then it is more visible, and you are sure that the code will compile

Same for the other tests

Reply:
I will add the in a near future, this is an intermediate PR. I prefer to do it with a comment